File: CS-Examples/05_Outlook Storage File/AddMessage.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Email;
using Spire.Email.Outlook;

namespace AddMessage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Load PST file
            Spire.Email.Outlook.OutlookFile olf = new OutlookFile(@"..\..\..\..\..\..\Data\Sample.pst");
            //Load Outlook MSG file
            OutlookItem item = new OutlookItem();
            item.LoadFromFile(@"..\..\..\..\..\..\Data\Sample.msg");
            //Select the "Inbox" folder
            OutlookFolder inboxFolder = olf.RootOutlookFolder.GetSubFolder("Inbox");
            //Add the MSG to "Inbox" folder
            inboxFolder.AddItem(item);
            MessageBox.Show("Completed");
        }
    }
}

File: VB-Examples/05_Outlook Storage File/AddMessage.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Email
Imports Spire.Email.Outlook

Namespace AddMessage
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Load PST file
            Dim olf As Spire.Email.Outlook.OutlookFile = New OutlookFile("..\..\..\..\..\..\Data\Sample.pst")
			'Load Outlook MSG file
			Dim item As New OutlookItem()
            item.LoadFromFile("..\..\..\..\..\..\Data\Sample.msg")
			'Select the "Inbox" folder
			Dim inboxFolder As OutlookFolder = olf.RootOutlookFolder.GetSubFolder("Inbox")
			'Add the MSG to "Inbox" folder
            inboxFolder.AddItem(item)
            MessageBox.Show("Completed")
		End Sub
	End Class
End Namespace